### PR TITLE
PR push

### PR DESCRIPTION
**Pull request by scmountain**
up to mean/median/mode

### CODE REVIEW

**Comment by scmountain** [comment]
@case-eee @neight-allen just a quick push to show where i am on this. not finished yet but trying to move to next assignment to keep nose above water.

**Comment by scmountain** [comment]
median finished, 12 tests

File: lib/average.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+function getSum(total, num) {
+    return total + num;
+};
+
+function total(list){
+  return list.reduce(getSum)
+};
+
+function mean(list){
+  return (total(list) / list.length)
```
Comment (case-eee) [inline lib/average.js]: love that you're using `total` within this function 👍

File: lib/average.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        return (list[half-1] + list[half]) / 2.0;
+}
+
+function mode(list) {
+    var counter = {};
+    var mode = [];
+    var max = 0;
+    for (var i in list) {
+        if (!(list[i] in counter))
+            counter[list[i]] = 0;
```
Comment (case-eee) [inline lib/average.js]: watch your indentation here - it looks a little wonky here

File: lib/average.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            mode = [list[i]];
+        }
+    }
+    return mode;
+}
+
+exports.total = total
+exports.mean = mean
+exports.median = median
+exports.mode = mode
```
Comment (case-eee) [inline lib/average.js]: good work with this 👍

File: lib/rectangle.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
...
+  return rectangle.width * 2 + rectangle.height * 2
+};
+function diagonal(rectangle){
+  return Math.sqrt((rectangle.width * rectangle.width) + (rectangle.height * rectangle.height)).toFixed(2)
+};
+
+function square(rectangle){
+  if (rectangle.width == rectangle.height){
+  return true
+}
```
Comment (case-eee) [inline lib/rectangle.js]: again, watch your spacing. interviewers may judge you on your ability to follow spacing conventions (i know it seems a little silly but i've heard of it happening!)

File: lib/triangle.js [hunk 5]
```diff
@@ -0,0 +1,8 @@
+function isTriangle(a,b,c){
```
Comment (case-eee) [inline lib/triangle.js]: looks good!

**Pull request closed without merging**